### PR TITLE
Support MAC 1.0.3 and PHY 1.0.3 Rev A

### LINKED ISSUE #474: LoRaWAN 1.0.3 support
**Summary:** 

LoRaWAN 1.0.3 and Regional Parameters 1.0.3 Rev A support.

**Why do we need this?**

It's specification.

**What is already there? What do you see now?**

MAC 1.0.2 and PHY 1.0.2 Rev B, as well as MAC 1.1 and PHY 1.1 Rev A.

**What is missing? What do you want to see?**

MAC 1.0.3 and PHY 1.0.3 Rev A.

**How do you propose to implement this?**

Add the enum values and sanity check the version checking.

The MAC command `DeviceTimeReq`/`Ans` are already supported.

**Can you do this yourself and submit a Pull Request?**

Yes.

### PR DESCRIPTION
**Pull request by johanstokking**
**Summary:**

Closes #474 

**Changes:**

- Add MAC 1.0.3 and PHY 1.0.3 Rev A in API
- Apply more semantic version checks
- Support MAC 1.0.3 in JS

**Notes for Reviewers:**

See my review comment

### CODE REVIEW

File: pkg/networkserver/grpc_gsns.go [hunk 1]
```diff
@@ -633,7 +633,7 @@ func (ns *NetworkServer) handleUplink(ctx context.Context, up *ttnpb.UplinkMessa
 				case ttnpb.CID_LINK_ADR:
 					pld := cmd.GetLinkADRAns()
 					dupCount := 0
-					if stored.MACState.LoRaWANVersion == ttnpb.MAC_V1_0_2 {
+					if stored.MACState.LoRaWANVersion.Compare(ttnpb.MAC_V1_0_2) >= 0 {
```
Comment (johanstokking) [inline pkg/networkserver/grpc_gsns.go]: Assuming here that atomic application of multiple `LinkAdrReq` is enforced by 1.0.2 or higher.

Not sure why the previous check was exactly on 1.0.2; seems wrong.

**Comment by johanstokking** [comment]
FTR; tested back-to-back with Mbed OS 5.12.1 with the getting started scenarios.

File: pkg/networkserver/grpc_gsns.go (hunk 1 above)
Comment (rvolosatovs) [reply pkg/networkserver/grpc_gsns.go]: This was correct behavior. There must be no duplicates in 1.1+
2019-04-15-11:16:33-screenshot

The check should be `1.0.2 <= ver < 1.1` for proper 1.0.X support

File: pkg/networkserver/mac_link_adr.go [hunk 2]
```diff
@@ -114,14 +114,14 @@ func handleLinkADRAns(ctx context.Context, dev *ttnpb.EndDevice, pld *ttnpb.MACC
 		handler = handleMACResponse
 	}
 
-	if dev.MACState.LoRaWANVersion != ttnpb.MAC_V1_0_2 && dupCount != 0 {
+	if dev.MACState.LoRaWANVersion.Compare(ttnpb.MAC_V1_0_2) < 0 && dupCount != 0 {
```
Comment (rvolosatovs) [inline pkg/networkserver/mac_link_adr.go]: Incorrect https://github.com/TheThingsNetwork/lorawan-stack/pull/475/files#r275272343

File: pkg/networkserver/mac_link_adr.go [hunk 3]
```diff
@@ -114,14 +114,14 @@ func handleLinkADRAns(ctx context.Context, dev *ttnpb.EndDevice, pld *ttnpb.MACC
...
 		return errInvalidPayload
 	}
 
 	var n uint
 	var req *ttnpb.MACCommand_LinkADRReq
 	dev.MACState.PendingRequests, err = handler(ttnpb.CID_LINK_ADR, func(cmd *ttnpb.MACCommand) error {
-		if dev.MACState.LoRaWANVersion == ttnpb.MAC_V1_0_2 && n > dupCount+1 {
+		if dev.MACState.LoRaWANVersion.Compare(ttnpb.MAC_V1_0_2) >= 0 && n > dupCount+1 {
```
Comment (rvolosatovs) [inline pkg/networkserver/mac_link_adr.go]: IDEM

File: pkg/networkserver/grpc_gsns.go (hunk 1 above)
Comment (johanstokking) [reply pkg/networkserver/grpc_gsns.go]: Ok, please file issue

**Pull request merged**